### PR TITLE
feat(math): implement m:sSub subscript converter (SD-2373)

### LINKED ISSUE #2596: Math: implement m:sSub subscript converter (community)
## What

Implement the `m:sSub` (subscript) OMML → MathML converter.

**OMML**: `m:sSub` → `m:e` (base) + `m:sub` (subscript)<br>**MathML**: `<msub>` with two children (base, subscript)<br>**Spec**:  §22.1.2.101<br>**Difficulty**: Easy

## Example

Word: a₁ → `<m:sSub><m:e><m:r><m:t>a</m:t></m:r></m:e><m:sub><m:r><m:t>1</m:t></m:r></m:sub></m:sSub>`<br>MathML: `<msub><mi>a</mi><mn>1</mn></msub>`

## How to implement

1. Create `packages/layout-engine/painters/dom/src/features/math/converters/subscript.ts`
2. Export `convertSubscript: MathObjectConverter`
3. Register in `converters/index.ts` and `omml-to-mathml.ts` registry (`'m:sSub': convertSubscript`)

See `converters/fraction.ts` for the pattern.

## Test Document

Download the test file and upload it to the SuperDoc dev app to verify your implementation renders correctly:

[sd-2373-subscript.docx](https://uploads.linear.app/92b189d9-e15d-4f28-8a87-d9ca147ba33f/baa0f695-69fd-44cc-ae25-29bf9aba0619/228451de-deba-4484-bcb0-95badade1761?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzkyYjE4OWQ5LWUxNWQtNGYyOC04YTg3LWQ5Y2ExNDdiYTMzZi9iYWEwZjY5NS02OWZkLTQ0Y2MtYWUyNS0yOWJmOWFiYTA2MTkvMjI4NDUxZGUtZGViYS00NDg0LWJjYjAtOTViYWRhZGUxNzYxIiwiaWF0IjoxNzc0NjQ4NDAxLCJleHAiOjE4MDYyMTg5NjF9.LjxBBMYeb8mLcBgMq1C27NKQG3Sq3kKX1sU2ijoIgn0)

---

**Want to work on this?** Check the [Contributing Guide](<https://github.com/superdoc-dev/superdoc/blob/main/CONTRIBUTING.md>) to get started. Comment on this issue to let us know you're picking it up.

*Ref:* [SD-2373](https://linear.app/superdocworkspace/issue/SD-2373/math-implement-mssub-subscript-converter-community)

### PR DESCRIPTION
**Pull request by gpardhivvarma**
## Summary

- Add OMML `m:sSub` → MathML `<msub>` converter following the `fraction.ts` pattern
- Move `m:f` from the "not yet implemented" registry section to "implemented" where it belongs
- 3 tests: basic conversion (a₁), properties element ignored, missing `m:sub` graceful degradation

## Test plan

- [x] `pnpm --filter @superdoc/painter-dom test` — `omml-to-mathml.test.ts` passes (17 tests)
- [ ] Upload [sd-2373-subscript.docx](https://uploads.linear.app/92b189d9-e15d-4f28-8a87-d9ca147ba33f/baa0f695-69fd-44cc-ae25-29bf9aba0619/228451de-deba-4484-bcb0-95badade1761) to dev app and verify subscript renders correctly

Closes #2596

### CODE REVIEW

File: packages/layout-engine/painters/dom/src/features/math/converters/subscript.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * @spec ECMA-376 §22.1.2.101
+ */
+export const convertSubscript: MathObjectConverter = (node, doc, convertChildren) => {
+  const elements = node.elements ?? [];
+  const base = elements.find((e) => e.name === 'm:e');
+  const sub = elements.find((e) => e.name === 'm:sub');
+
+  const msub = doc.createElementNS(MATHML_NS, 'msub');
+  msub.appendChild(convertChildren(base?.elements ?? []));
```
Comment (caio-pizzol) [inline packages/layout-engine/painters/dom/src/features/math/converters/subscript.ts]: when the subscript has multiple parts (like `n+1`), they end up as separate children instead of being grouped. `fraction.ts` has the same issue — we'll fix all converters together in a follow-up.

not blocking — single-part subscripts work fine.

**Review by caio-pizzol — COMMENTED** [review]
@gpardhivvarma nice work — follows the existing pattern well and the spec reference is correct.

simple subscripts like `a₁` render correctly. when the base or subscript has multiple parts (like `x_{n+1}`), they don't get grouped properly and the rendering breaks. `fraction.ts` has the same issue — `bar.ts` already handles it. we'll fix all converters together in a follow-up.

tests look good.

i made a .docx with 10 subscript variations:

[sd-2373-subscript-edge-cases-v2.docx](https://github.com/user-attachments/files/26360989/sd-2373-subscript-edge-cases-v2.docx)


| # | expression | works? |
|---|-----------|--------|
| 1 | a₁ | yes |
| 2 | x\_{n+1} | no — 4 children instead of 2 |
| 3 | (a+b)\_n | no — 4 children instead of 2 |
| 4 | (a+b)\_{n+1} | no — 6 children instead of 2 |
| 5 | x\_{i+j+k+1} | no — 8 children instead of 2 |
| 6 | a\_{b\_c} (nested) | yes |
| 7 | x\_{a/b} (fraction) | yes |
| 8 | y₀ (with properties) | yes |
| 9 | a₁+b₂ (multiple) | yes |
| 10 | x\_{} (empty) | no — 1 child instead of 2 |

Word: 
image

SuperDoc:
image

simple cases (1, 6-9) look the same. multi-part cases (2-5) render incorrectly in SuperDoc.

left an inline comment.

File: packages/layout-engine/painters/dom/src/features/math/converters/subscript.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  msub.appendChild(convertChildren(sub?.elements ?? []));
```
[the remark this answers is not in the document]
Comment (caio-pizzol) [reply packages/layout-engine/painters/dom/src/features/math/converters/subscript.ts]: @gpardhivvarma confirmed — we'll fix this across all converters in a follow-up.

[the remark this answers is not in the document]
Comment (gpardhivvarma) [reply packages/layout-engine/painters/dom/src/features/math/converters/subscript.ts]: Fixed — operands are now wrapped in `<mrow>` before appending, matching the pattern in `bar.ts`. Multi-part expressions like `x_{n+1}` now produce exactly 2 children in `<msub>`. Added a test for this case too.

File: packages/layout-engine/painters/dom/src/features/math/converters/subscript.ts (hunk 1 above)
Comment (gpardhivvarma) [reply packages/layout-engine/painters/dom/src/features/math/converters/subscript.ts]: Thanks for the thorough testing! This is now fixed in the latest push — also wrapped `<mrow>` around fraction operands while at it since it had the same issue.

**Review by caio-pizzol — APPROVED** [review]
@gpardhivvarma the grouping fix works — tested 15 edge cases and everything renders correctly now. you also fixed the same issue in `fraction.ts` which is a nice bonus.

uploaded the edge case test file to our test corpus so we catch regressions going forward.

approving.
